plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ConnectionWrapper.java

```diff
@@ -237,14 +243,23 @@ protected void handleAdd(ProjectWrapper projectWrapper, IResource newResource) {
 
 	protected void handleRemove(ProjectWrapper projectWrapper, IResource oldResource) {
 		resourceCache.remove(oldResource);
-		Collection<IResource> resources = resourceCache.getResources(oldResource.getProject().getNamespace());
-		projectWrapper.updateWithResources(resources);
+		WatchManager.getInstance().stopWatch(projectWrapper.getWrapped(), projectWrapper.getParent().getWrapped(), oldResource.getKind());
+		if (oldResource instanceof IProject) {
+			synchronized(projects) {
+				projects.remove(oldResource.getName());
+				resourceCache.flush(oldResource.getName());
+				fireChanged();

```

Comment:
this could be a substantial bottleneck since we block other threads from getting processed even when we roundtrip to the UI. Is this really necessary?